geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/CreateDataSourceCommandDUnitTest.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.geode.cache.configuration.JndiBindingsType.JndiBinding;
+import org.apache.geode.distributed.internal.InternalConfigurationPersistenceService;
+import org.apache.geode.internal.jndi.JNDIInvoker;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.test.dunit.IgnoredException;
+import org.apache.geode.test.dunit.SerializableRunnableIF;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+import org.apache.geode.test.junit.rules.VMProvider;
+
+public class CreateDataSourceCommandDUnitTest {
+
+  private static MemberVM locator, server1, server2;
+
+  @ClassRule
+  public static ClusterStartupRule cluster = new ClusterStartupRule();
+
+  @ClassRule
+  public static GfshCommandRule gfsh = new GfshCommandRule();
+
+
+  @BeforeClass
+  public static void before() throws Exception {
+    locator = cluster.startLocatorVM(0);
+    server1 = cluster.startServerVM(1, "group1", locator.getPort());
+    server2 = cluster.startServerVM(2, "group1", locator.getPort());
+
+    gfsh.connectAndVerify(locator);
+  }
+
+  @Test
+  public void testCreateDataSource() throws Exception {
+    VMProvider.invokeInEveryMember(
+        () -> assertThat(JNDIInvoker.getNoOfAvailableDataSources()).isEqualTo(0), server1, server2);
+
+    String URL = "jdbc:derby:newDB;create=true";
+    // create the data-source
+    gfsh.executeAndAssertThat(
+        "create data-source --name=jndi1 --username=myuser --password=mypass --pooled=false --url=\""
+            + URL + "\"")
+        .statusIsSuccess().tableHasColumnOnlyWithValues("Member", "server-1", "server-2");
+
+    // verify cluster config is updated
+    locator.invoke(() -> {
+      InternalConfigurationPersistenceService ccService =
+          ClusterStartupRule.getLocator().getConfigurationPersistenceService();
+      CacheConfig cacheConfig = ccService.getCacheConfig("cluster");
+      List<JndiBinding> jndiBindings = cacheConfig.getJndiBindings();
+      assertThat(jndiBindings.size()).isEqualTo(1);
+      JndiBinding jndiBinding = jndiBindings.get(0);
+      assertThat(jndiBinding.getJndiName()).isEqualTo("jndi1");
+      assertThat(jndiBinding.getUserName()).isEqualTo("myuser");
+      assertThat(jndiBinding.getPassword()).isEqualTo("mypass");
+      assertThat(jndiBinding.getConnectionUrl()).isEqualTo(URL);
+      assertThat(jndiBinding.getType()).isEqualTo("SimpleDataSource");
+    });
+
+    // verify datasource exists
+    VMProvider.invokeInEveryMember(
+        () -> assertThat(JNDIInvoker.getNoOfAvailableDataSources()).isEqualTo(1), server1, server2);
+
+    // bounce server1
+    server1.stop(false);
+    server1 = cluster.startServerVM(1, locator.getPort());
+
+    // verify it has recreated the datasource from cluster config
+    server1.invoke(() -> {
+      assertThat(JNDIInvoker.getNoOfAvailableDataSources()).isEqualTo(1);
```

Comment:
Can we verify the name of the data source; to make sure its the one that got created.

Reply:
done